lib/schemes/oauth2.js

```diff
@@ -52,6 +52,8 @@ export default class Oauth2Scheme {
     // Handle callbacks on page load
     const redirected = await this._handleCallback()
 
+    // TODO: determine if it is necessary to pull user info on EACH and EVERY page request?
```

Comment:
Indeed it is. HTTP is stateless by nature! So for each SSR request, we need to either restore user from cookies (which implies transferring an object 2x per request) or fetching from remote.